src/Draco.RedGreenTree.SourceGeneration/RedTreeSourceGenerator.cs

```diff
@@ -0,0 +1,42 @@
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace Draco.RedGreenTree.SourceGeneration;
+
+[Generator]
+public sealed class RedTreeSourceGenerator : IIncrementalGenerator
+{
+    public void Initialize(IncrementalGeneratorInitializationContext context)
+    {
+        const string attributeName = "Draco.RedGreenTree.RedTreeAttribute";
+
```

Comment:
Shouldn't we also inject the attribute source code?

Reply:
Fair point, although I think that'd have to be its own separate generator which only purpose is checking whether the attribute exists and otherwise adding it.

Reply:
I mean we could just unconditionally add all the attributes in a separate generator. I don't think there's any reason not to